src/ImageSharp/Common/Helpers/Vector128Utilities.cs

```diff
@@ -193,13 +193,60 @@ public static Vector128<int> ConvertToInt32RoundToEven(Vector128<float> vector)
             return AdvSimd.ConvertToInt32RoundToEven(vector);
         }
 
-        Vector128<float> sign = vector & Vector128.Create(-0.0f);
-        Vector128<float> val_2p23_f32 = sign | Vector128.Create(8388608.0f);
+        Vector128<float> sign = vector & Vector128.Create(-0F);
```

Comment:
No general helper exists for .NET 9+ here...

It's something we could probably pattern recognize as `ConvertToInt32(Round(vector))` though. If it's a pattern you'd like to see us optimize, logging a suggestion on dotnet/runtime would be beneficial. Alternatively some `ConvertToInt32(vector, MidpointRounding)` API could be provided as well and an API suggestion could be opened.

Reply:
I'll have a look at creating an issue. Thanks.